File: NodeJs/src/mappers/extract.js
module.exports = (app) => {
  const View = app.models.view.extract;

  class ExtractMapper {
    static async modelToViewModel(model) {
      try {
        if (!model) return null;

        const view = Object.assign({}, View);

        view.DateLaunch = model.dataLancamentoContaCorrenteCliente;
        view.Description = model.lancamentoContaCorrenteCliente.nomeTipoOperacao;
        view.Number = model.numeroEvento;
        view.Location = model.lancamentoContaCorrenteCliente.nomeSituacaoRemessa;
        view.DateConfirmation = model.dataEfetivaLancamento;
        view.BankName = model.nomeBanco.trim();
        view.BankAgency = model.lancamentoContaCorrenteCliente.dadosDomicilioBancario.numeroAgencia;
        view.BankAccount = model.lancamentoContaCorrenteCliente.dadosDomicilioBancario.numeroContaCorrente;
        view.BankDetailsComplete = `${model.nomeBanco.trim()} Ag ${model.lancamentoContaCorrenteCliente.dadosDomicilioBancario.numeroAgencia} CC ${model.lancamentoContaCorrenteCliente.dadosDomicilioBancario.numeroContaCorrente}`;
        view.FinalValue = model.valorLancamentoRemessa;
        view.FinalValueLocal = model.valorLancamentoRemessa.toLocaleString('pt-br', { style: 'currency', currency: 'BRL' });

        return view;
      } catch (error) {
        throw new Error(`ExtractMapper - ModelToViewModel : ${error}`);
      }
    }

    static async modelsToViewsModel(models) {
      try {
        if (!models) return null;

        const views = [];

        models.listaControleLancamento.forEach(async (model) => {
          const view = await this.modelToViewModel(model);

          views.push(view);
        });

        return views;
      } catch (error) {
        throw new Error(`ExtractMapper - ModelsToViewsModel : ${error}`);
      }
    }
  }

  return ExtractMapper;
};

File: AspNetCore/TestResource.Domain/Model/Extract.cs
using System.Collections.Generic;

namespace TestResource.Domain.Model
{
    public class Extract
    {
        public TotalControleLancamento TotalControleLancamento { get; set; }
        public List<ListaControleLancamento> ListaControleLancamento { get; set; }
        public int Indice { get; set; }
        public int TamanhoPagina { get; set; }
        public int TotalElements { get; set; }
    }

    public class TotalControleLancamento
    {
        public int QuantidadeLancamentos { get; set; }
        public int QuantidadeRemessas { get; set; }
        public double ValorLancamentos { get; set; }
    }

    public class DadosDomicilioBancario
    {
        public int CodigoBanco { get; set; }
        public int NumeroAgencia { get; set; }
        public string NumeroContaCorrente { get; set; }
    }

    public class LancamentoContaCorrenteCliente
    {
        public object NumeroRemessaBanco { get; set; }
        public string NomeSituacaoRemessa { get; set; }
        public DadosDomicilioBancario DadosDomicilioBancario { get; set; }
        public string NomeTipoOperacao { get; set; }
        public List<object> DadosAnaliticoLancamentoFinanceiroCliente { get; set; }
    }

    public class ListaControleLancamento
    {
        public LancamentoContaCorrenteCliente LancamentoContaCorrenteCliente { get; set; }
        public string DataEfetivaLancamento { get; set; }
        public string DataLancamentoContaCorrenteCliente { get; set; }
        public int NumeroEvento { get; set; }
        public string DescricaoGrupoPagamento { get; set; }
        public string CodigoIdentificadorUnico { get; set; }
        public string NomeBanco { get; set; }
        public int QuantidadeLancamentoRemessa { get; set; }
        public string NumeroRaizCnpj { get; set; }
        public string NumeroSufixoCnpj { get; set; }
        public double ValorLancamentoRemessa { get; set; }
        public object DateLancamentoContaCorrenteCliente { get; set; }
        public object DateEfetivaLancamento { get; set; }
    }
}